src/libertem/udf/holography.py

```diff
@@ -0,0 +1,244 @@
+#
+# You should have received a copy of the GNU General Public License
+# along with  HyperSpy.  If not, see <http://www.gnu.org/licenses/>.
+# 
+# Copyright 2019 The LiberTEM developers
+# 
+#  LiberTEM is distributed under the terms of the GNU General 
+# Public License as published by the Free Software Foundation,
+# version 3 of the License.
+# see: https://github.com/LiberTEM/LiberTEM
+
+import numpy as np
+from numpy.fft import fftshift, fft2, ifft2
+from libertem.udf import UDF
+
+
+def freq_array(shape, sampling=(1., 1.)):
+    """
+    Makes up a frequency array.
+
+    Parameters
+    ----------
+    shape : (int, int)
+        The shape of the array.
+    sampling: (float, float), optional, (Default: (1., 1.))
+        The sampling rates of the array.
+    Returns
+    -------
+        Array of the frequencies.
+    """
+    f_freq_1d_y = np.fft.fftfreq(shape[0], sampling[0])
+    f_freq_1d_x = np.fft.fftfreq(shape[1], sampling[1])
+    f_freq_mesh = np.meshgrid(f_freq_1d_x, f_freq_1d_y)
+    f_freq = np.hypot(f_freq_mesh[0], f_freq_mesh[1])
+
+    return f_freq
+
+
+def aperture_function(r, apradius, rsmooth):
+    """
+    A smooth aperture function that decays from apradius-rsmooth to apradius+rsmooth.
+    Parameters
+    ----------
+    r : ndarray
+        Array of input data (e.g. frequencies)
+    apradius : float
+        Radius (center) of the smooth aperture. Decay starts at apradius - rsmooth.
+    rsmooth : float
+        Smoothness in halfwidth. rsmooth = 1 will cause a decay from 1 to 0 over 2 pixel.
+    Returns
+    -------
+        2d array containing aperture
+    """
+
+    return 0.5 * (1. - np.tanh((np.absolute(r) - apradius) / (0.5 * rsmooth)))
+
+
+class HoloReconstructUDF(UDF):
+    """
+    Reconstruct off-axis electron holograms using Fourier-based method
+    Running `Context.run_udf` on instance of this class will a complex electron wave.
+    Use `wave` key to access raw data.
+
+    Examples
+    --------
+    >>> shape = tuple(dataset.shape.sig)
+    >>> sb_position = [2, 3]
+    >>> sb_size = 4.4
+    >>> holo_job = HoloReconstructUDF(out_shape=shape,
+    ...                               sb_position=sb_position,
+    ...                               sb_size=sb_size)
+    >>> wave = ctx.run_udf(dataset=dataset, udf=holo_job)['wave'].data
+    """
+
+    def __init__(self,
+                 out_shape,
+                 sb_position,
+                 sb_size,
+                 sb_smoothness=.05,
+                 precision=True):
+        """
+        out_shape : (int, int)
+            Shape of the outputted complex wave image
+
+        sb_position : tuple, or vector
+            Coordinates of sideband position with respect to non-shifted FFT of a hologram
+
+        sb_size : float
+            Radius of side band filter in pixels
+
+        sb_smoothness : float, optional (Default: 0.05)
+            Fraction of `sb_size` over which the edge of the filter aperture to be smoothed
```

Comment:
In the documentation you wrote percentage, here fraction, I guess this docstring is right?